### PR TITLE
Bump deps of knative-serving to 0.13, openshift to 4.3 and use servingv1 API.

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title, this is long overdue and broken out of #178 to make that PR not overwhelmed with dep updates.

### CODE REVIEW

File: test/Gopkg.toml [hunk 1]
```diff
@@ -15,44 +15,48 @@ required = [
...
+  version = "kubernetes-1.16.4"
 
 [[constraint]]
-  branch = "master"
   name = "knative.dev/serving-operator"
+  branch = "release-0.13"
 
 [[constraint]]
-  branch = "master"
   name = "github.com/knative/test-infra"
```
Comment (cardil) [inline test/Gopkg.toml:41]: Let's also change a `github.com/knative/test-infra` to `knative.dev/test-infra`. I think I can provide a commit for that.

Comment (markusthoemmes) [reply test/Gopkg.toml:41]: sure, should be simple. On it.

Comment (markusthoemmes) [reply test/Gopkg.toml]: Pushed.

File: test/Gopkg.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Force dep to vendor the code generators, which aren't imported just used at dev time.
 required = [
-  "github.com/knative/test-infra/scripts",
-  "github.com/knative/test-infra/tools/dep-collector",
+  "knative.dev/test-infra/scripts",
+  "knative.dev/test-infra/tools/dep-collector",
```
Comment (navidshaikh) [inline test/Gopkg.toml:4]: ``
[prune]
  go-tests = true
  non-go = true
  unused-packages = true
[[prune.project]]
  name = "github.com/knative/test-infra"
  non-go = false
```

The prune setting isnt applied causing the script `test/vendor/knative.dev/test-infra/scripts/e2e-tests.sh` to be deleted and test fails
```
./test/lib.bash: line 42: go_test_e2e: command not found
  ERROR 14:37:09.808 Tests have failures!
```

need to change
```
[[prune.project]]
  name = "github.com/knative/test-infra"
```
part to `knative.dev/test-infra`

Comment (markusthoemmes) [reply test/Gopkg.toml:4]: ugh, thanks!

**Pull request closed without merging**